src/transformers/models/ict/modeling_ict.py

```diff
@@ -0,0 +1,864 @@
+    https://github.com/dxyang/StyleTransfer/blob/master/utils.py
+    """
+
+    def __init__(self, weights=None):
+        super().__init__()
+        self.add_module("vgg", VGG19())
+        self.criterion = torch.nn.L1Loss()
+        self.weights = weights if weights is not None else [1.0, 1.0, 1.0, 1.0, 1.0]
+
+    def __call__(self, x, y):
+        # Compute features
+        x_vgg, y_vgg = self.vgg(x), self.vgg(y)
+
+        content_loss = 0.0
+        content_loss += self.weights[0] * self.criterion(x_vgg["relu1_1"], y_vgg["relu1_1"])
+        content_loss += self.weights[1] * self.criterion(x_vgg["relu2_1"], y_vgg["relu2_1"])
+        content_loss += self.weights[2] * self.criterion(x_vgg["relu3_1"], y_vgg["relu3_1"])
+        content_loss += self.weights[3] * self.criterion(x_vgg["relu4_1"], y_vgg["relu4_1"])
+        content_loss += self.weights[4] * self.criterion(x_vgg["relu5_1"], y_vgg["relu5_1"])
+
+        return content_loss
+
+
+class IctGuidedUpsampler(IctPretrainedModel):
+    def __init__(self, config: IctConfig):
+        super().__init__(config)
+
+        self.generator = IctInpaintGenerator(config)
+        self.adversarial_loss = IctAdversarialLoss()
+        self.l1_loss = nn.L1Loss()
+        self.perceptual_loss = IctPerceptualLoss()
+        self.style_loss = IctStyleLoss()
+        self.output_height = config.output_height
+        self.output_width = config.output_width
+
+        self.post_init()
+
+    # modified from https://github.com/raywzy/ICT/blob/59dd12d374d47cdf0dce90923017ca3657e6aa0b/Guided_Upsample/src/dataset_my.py#L203-L209
+    # and https://github.com/raywzy/ICT/blob/59dd12d374d47cdf0dce90923017ca3657e6aa0b/Guided_Upsample/src/dataset_my.py#L183-L186
+    def resize(self, img: torch.Tensor, target_height: int, target_width: int):
+        img = img.cpu().detach().numpy()
+        img_height, img_width = img.shape[0:2]
+
+        if img_height != img_width:
+            # Center crop
+            side = min(img_height, img_width)
+            j = (img_height - side) // 2
+            i = (img_width - side) // 2
+            img = img[j : j + side, i : i + side, ...]
+        img = img.astype(np.uint8)
+        np_img = Image.fromarray(img)
+        resized_np_img = np_img.resize((target_height, target_width), resample=Image.BICUBIC)
+
+        return F.to_tensor(resized_np_img).float()
+
+    # modified from https://github.com/raywzy/ICT/blob/59dd12d374d47cdf0dce90923017ca3657e6aa0b/Guided_Upsample/src/models.py#L165-L183
+    def forward(self, images: List[torch.Tensor], edges: List[torch.Tensor], masks: List[torch.Tensor]):
+        images = torch.stack([self.resize(image, self.output_height, self.output_width) for image in images])
+        masks = torch.stack([self.resize(mask, self.output_height, self.output_width) for mask in masks])
+        edges = torch.stack([self.resize(edge, self.output_height, self.output_width) for edge in edges])
```

Comment:
Do we need to iterate over the batches like this? I thought all the images and masks have the same dimensions as they're batched together